### PR TITLE
#155: Added a Notebook Link Checker to Github Actions

### LINKED ISSUE #155: Add link checker for notebooks

### PR DESCRIPTION
**Pull request by tkilias**
Fixes #155 
Fixes #129

### CODE REVIEW

**Review by ckunki — CHANGES_REQUESTED** [review]
Please update changes file
Otherwise I am happy to approve

File: .github/workflows/check_notebook_links.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+     - uses: actions/checkout@v3
+       with:
+         fetch-depth: 0
+     - name: Setup Python & Poetry Environment
+       uses: ./.github/actions/prepare_poetry_env
+
+     - name: Run build ai-lab tests
+       run: >
+         poetry run pytest --check-links exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/
```
Comment (ahsimb) [inline .github/workflows/check_notebook_links.yaml:21]: Do you think it will search deep for all files in the directory?

Comment (ahsimb) [reply .github/workflows/check_notebook_links.yaml:21]: It's probably easier to give it a try and see what will happen.

Comment (tkilias) [reply .github/workflows/check_notebook_links.yaml:21]: from the CI run

```

exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/cloud_store_config.ipynb . [  0%]
..                                                                       [  1%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/main_config.ipynb . [  2%]
                                                                         [  2%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/start.ipynb . [  2%]
........                                                                 [  7%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/cloud/01_import_data.ipynb . [  8%]
...                                                                      [ 10%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/data/data_abalone.ipynb . [ 10%]
..                                                                       [ 11%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/data/data_content.ipynb . [ 12%]
.                                                                        [ 13%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/data/data_telescope.ipynb . [ 13%]
..                                                                       [ 14%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/sagemaker/sme_deploy_model.ipynb . [ 15%]
.......                                                                  [ 19%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/sagemaker/sme_init.ipynb . [ 19%]
....                                                                     [ 22%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/sagemaker/sme_introduction.ipynb . [ 22%]
.                                                                        [ 23%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/sagemaker/sme_train_model.ipynb . [ 23%]
.......                                                                  [ 27%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/sklearn/sklearn_introduction.ipynb . [ 28%]
........                                                                 [ 33%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/sklearn/sklearn_predict_abalone.ipynb . [ 33%]
.....                                                                    [ 36%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/sklearn/sklearn_predict_telescope.ipynb . [ 37%]
.....                                                                    [ 40%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/sklearn/sklearn_predict_udf.ipynb . [ 41%]
..                                                                       [ 42%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/sklearn/sklearn_train_abalone.ipynb . [ 42%]
.......                                                                  [ 47%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/sklearn/sklearn_train_telescope.ipynb . [ 47%]
........                                                                 [ 52%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/slc/script-languages.ipynb . [ 52%]
......                                                                   [ 56%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/transformers/masked_modelling.ipynb . [ 57%]
........                                                                 [ 61%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/transformers/question_answering.ipynb . [ 61%]
.......                                                                  [ 66%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/transformers/sequence_classification.ipynb . [ 66%]
.......                                                                  [ 70%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/transformers/te_init.ipynb . [ 71%]
.                                                                        [ 72%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/transformers/te_introduction.ipynb . [ 72%]
.........                                                                [ 77%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/transformers/text_generation.ipynb . [ 78%]
......                                                                   [ 82%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/transformers/token_classification.ipynb . [ 82%]
.......                                                                  [ 86%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/transformers/translation.ipynb . [ 87%]
.......                                                                  [ 91%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/transformers/zero_shot_classification.ipynb . [ 92%]
.......                                                                  [ 96%]
exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/transformers/utils/model_retrieval.ipynb . [ 97%]
```

Comment (ahsimb) [reply .github/workflows/check_notebook_links.yaml:21]: Looks convincing.

**Review by ckunki — APPROVED** [review]

**Pull request merged**